tasks/configure_agent.yml

```diff
@@ -0,0 +1,38 @@
+---
+- name: ensure python-yaml is installed
+  package:
+    name: python-yaml
+    state: "{{ nrinfragent_state }}"
+  when:
+    - nrinfragent_state != "absent"
+    - nrinfragent_os_name != 'windows'
+    - ansible_python_version is version('3.0', '<')
+
+- name: ensure python3-yaml is installed
+  package:
+    name: python3-yaml
+    state: "{{ nrinfragent_state }}"
+  when:
+    - nrinfragent_state != "absent"
+    - nrinfragent_os_name != 'windows'
+    - ansible_python_version is version('3.0', '>=')
+
+- name: setup agent config *NIX
+  yedit:
+    src: "{{ nrinfragent_config_path }}"
+    key: "{{ item.key }}"
+    value: "{{ item.value }}"
+  loop: "{{ lookup('ansible.utils.to_paths', nrinfragent_config) | dict2items }}"
```

Comment:
Info regarding the requirement of ansible.utils would be useful. It crashed on this step when I was running it locally.

Reply:
@JakubKotkowiak with the change to use `merge_yaml.py` (see next comment), this is no longer required.